pool/pool_manager.gno

```diff
@@ -30,6 +51,87 @@ func init() {
 	feeAmountTickSpacing[10000] = 200 // 1%
 }
 
+type poolInput struct {
+	token0Path, token1Path string
+	fee                    uint32
+	sqrtPriceX96           *u256.Uint
+	tickSpacing            int32
+}
+
+func newPoolInput(
+	token0Path string,
+	token1Path string,
+	fee uint32,
+	sqrtPriceX96 string,
+) *poolInput {
+	price := u256.MustFromDecimal(sqrtPriceX96)
+	tickSpacing := feeAmountTickSpacing.Get(fee)
+	return &poolInput{
+		token0Path: token0Path,
+		token1Path: token1Path,
+		fee:        fee,
+		sqrtPriceX96: price,
+		tickSpacing: tickSpacing,
+	}
+}
+
+func (p *poolInput) isSameTokenPath() bool {
+	return p.token0Path == p.token1Path
+}
+
+func (p *poolInput) isInOrder() bool {
+	if strings.Compare(p.token0Path, p.token1Path) > 0 {
```

Comment:
` > 0` vs `== 1`, and rule for this ?

Reply:
There is no strict rule for this, but I prefer `> 0` because this function determines the value relative to 0.